include/wil/stl.h

```diff
@@ -96,7 +96,7 @@ namespace wil
                 return E_OUTOFMEMORY;
             }
 
-            wchar_t* buffer() { return &m_value[0]; }
+            wchar_t* buffer() { return const_cast<wchar_t *>((m_value).data()); }

```

Comment:
This is less clear than the original code; I need to think about why the `const_cast` is here and what the implications might be, etc.